interfaces/system_key.go

```diff
@@ -73,19 +73,21 @@ type systemKey struct {
 var (
 	isHomeUsingNFS  = osutil.IsHomeUsingNFS
 	mockedSystemKey *systemKey
+	osReadlink      = os.Readlink
 )
 
 func findSnapdPath() (string, error) {
 	snapdPath := filepath.Join(dirs.DistroLibExecDir, "snapd")
 
 	// find the right snapdPath by looking if we are re-execing or not
-	exe, err := os.Readlink("/proc/self/exe")
+	exe, err := osReadlink("/proc/self/exe")
 	if err != nil {
 		return "", err
 	}
 
 	if strings.HasPrefix(exe, dirs.SnapMountDir) {
-		return filepath.Join(dirs.SnapMountDir, "core/current/usr/lib/snapd/snapd"), nil
+		prefix := strings.Split(exe, "/usr/")[0]
```

Comment:
Hmm, what is this doing?

Reply:
It just a nice way to get the prefix snap dir of the current running exe, here is an example: https://play.golang.org/p/NwyFxVL6OeL

Reply:
Maybe a comment would be useful here:
```
// reexecd' snapd may be coming from either 'core' or 'snapd' snaps, find 
// the local prefix to the snap
// /snap/snapd/123/usr/bin/snap       -> /snap/snapd/123
// /snap/core/234/usr/lib/snapd/snapd -> /snap/core/234
```